rtgui/batchqueue.cc

```diff
@@ -161,328 +160,220 @@ bool BatchQueue::keyPressed (GdkEventKey* event)
-            int tiffBits   = options.saveFormat.tiffBits,    tiffUncompressed = options.saveFormat.tiffUncompressed;
-            int saveParams = options.saveFormat.saveParams;
-            int forceFormatOpts = options.forceFormatOpts;
-
-            Glib::ustring currLine(buffer);
-            int a = 0;
-
-            if (currLine.rfind('\n') != Glib::ustring::npos) {
-                a++;
-            }
-
-            if (currLine.rfind('\r') != Glib::ustring::npos) {
-                a++;
-            }
-
-            if (a) {
-                currLine = currLine.substr(0, currLine.length() - a);
-            }
-
-            // Looking for the image's full path
-            pos = currLine.find('|');
-
-            if (pos != Glib::ustring::npos) {
-                source = currLine.substr(0, pos);
-                currLine = currLine.substr(pos + 1);
-
-                // Looking for the procparams' full path
-                pos = currLine.find('|');
-
-                if (pos != Glib::ustring::npos) {
-                    paramsFile = currLine.substr(0, pos);
-                    currLine = currLine.substr(pos + 1);
+        // skipping the first row
+        std::getline (file, row);
 
-                    // Looking for the full output path; if empty, it'll use the template string
-                    pos = currLine.find('|');
+        while (std::getline (file, row)) {
 
-                    if (pos != Glib::ustring::npos) {
-                        outputFile = currLine.substr(0, pos);
-                        currLine = currLine.substr(pos + 1);
+            std::istringstream line (row);
 
-                        // No need to bother reading the last options, they will be ignored if outputFile is empty!
-                        if (!outputFile.empty()) {
+            values.clear ();
 
-                            // Looking for the saving format
-                            pos = currLine.find('|');
-
-                            if (pos != Glib::ustring::npos) {
-                                saveFmt = currLine.substr(0, pos);
-                                currLine = currLine.substr(pos + 1);
-
-                                // Looking for the jpeg quality
-                                pos = currLine.find('|');
-
-                                if (pos != Glib::ustring::npos) {
-                                    jpegQuality = atoi(currLine.substr(0, pos).c_str());
-                                    currLine = currLine.substr(pos + 1);
-
-                                    // Looking for the jpeg subsampling
-                                    pos = currLine.find('|');
-
-                                    if (pos != Glib::ustring::npos) {
-                                        jpegSubSamp = atoi(currLine.substr(0, pos).c_str());
-                                        currLine = currLine.substr(pos + 1);
+            while (std::getline(line, column, '|')) {
```

Comment:
Yeah, one does splitting using a separator so often, even the C++ standard library had to add something to make it simpler. :-D